nc-news/src/components/ArticleListByTopic.jsx

```diff
@@ -1,53 +1,47 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
 import * as api from "../api";
 
-function ArticlesListByTopic() {
+function ArticleListByTopic() {
+  const { topic } = useParams();
   const [articles, setArticles] = useState([]);
-  const [topics, setTopics] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const { topic } = useParams();
 
   useEffect(() => {
     setIsLoading(true);
     api.getArticles().then((articlesAPI) => {
       setArticles(articlesAPI);
       setIsLoading(false);
     });
-  }, []);
+  }, [topic]);

```

Comment:
you're not passing the topic to the back end

Because you're filtering only the results from page 1 of ALL articles you're not showing all the articles for a topic!

You'll need to let your API do the filtering using a query